persistent-postgresql/Database/Persist/Postgresql.hs

```diff
@@ -2039,3 +2064,151 @@ postgresMkColumns :: [EntityDef] -> EntityDef -> ([Column], [UniqueDef], [Foreig
+deriving instance (Real (BackendKey b)) => Real (BackendKey (RawPostgresql b))
+deriving instance (Enum (BackendKey b)) => Enum (BackendKey (RawPostgresql b))
+deriving instance (Bounded (BackendKey b)) => Bounded (BackendKey (RawPostgresql b))
+deriving instance (ToJSON (BackendKey b)) => ToJSON (BackendKey (RawPostgresql b))
+deriving instance (FromJSON (BackendKey b)) => FromJSON (BackendKey (RawPostgresql b))
+#endif
+
+
+#if MIN_VERSION_base(4,12,0)
+$(pure [])
+
+makeCompatibleInstances [t| forall b. Compatible b (RawPostgresql b) |]
+#else
+instance HasPersistBackend b => HasPersistBackend (RawPostgresql b) where
+    type BaseBackend (RawPostgresql b) = BaseBackend b
+    persistBackend = persistBackend . _persistentBackend
+
+instance (PersistStoreRead b) => PersistStoreRead (RawPostgresql b) where
+    get = withReaderT _persistentBackend . get
+    getMany = withReaderT _persistentBackend . getMany
+
+instance (PersistQueryRead b) => PersistQueryRead (RawPostgresql b) where
+    selectSourceRes filts opts = withReaderT _persistentBackend $ selectSourceRes filts opts
+    selectFirst filts opts = withReaderT _persistentBackend $ selectFirst filts opts
+    selectKeysRes filts opts = withReaderT _persistentBackend $ selectKeysRes filts opts
+    count = withReaderT _persistentBackend . count
+    exists = withReaderT _persistentBackend . exists
+
+instance (PersistQueryWrite b) => PersistQueryWrite (RawPostgresql b) where
+    updateWhere filts updates = withReaderT _persistentBackend $ updateWhere filts updates
+    deleteWhere = withReaderT _persistentBackend . deleteWhere
+
+instance (PersistUniqueRead b) => PersistUniqueRead (RawPostgresql b) where
+    getBy = withReaderT _persistentBackend . getBy
+
+instance (PersistStoreWrite b) => PersistStoreWrite (RawPostgresql b) where
+    insert = withReaderT _persistentBackend . insert
+    insert_ = withReaderT _persistentBackend . insert_
+    insertMany = withReaderT _persistentBackend . insertMany
+    insertMany_ = withReaderT _persistentBackend . insertMany_
+    insertEntityMany = withReaderT _persistentBackend . insertEntityMany
+    insertKey k = withReaderT _persistentBackend . insertKey k
+    repsert k = withReaderT _persistentBackend . repsert k
+    repsertMany = withReaderT _persistentBackend . repsertMany
+    replace k = withReaderT _persistentBackend . replace k
+    delete = withReaderT _persistentBackend . delete
+    update k = withReaderT _persistentBackend . update k
+    updateGet k = withReaderT _persistentBackend . updateGet k
+
+instance (PersistUniqueWrite b) => PersistUniqueWrite (RawPostgresql b) where
+    deleteBy = withReaderT _persistentBackend . deleteBy
+    insertUnique = withReaderT _persistentBackend . insertUnique
+    upsert rec = withReaderT _persistentBackend . upsert rec
+    upsertBy uniq rec = withReaderT _persistentBackend . upsertBy uniq rec
+    putMany = withReaderT _persistentBackend . putMany
+#endif
+
+makeLenses ''RawPostgresql
```

Comment:
Ah, dang, I see that this is copied over from the `sqlite` stuff. I wish I had noticed that :\

Let's not bring in the `microlens-th` dependency and derive these lenses. Instead, just expose the record constructor and labels directly. If someone wants to drive lenses for the type, they can do so (either with `Control.Lens` or `Lens.Micro.TH`).

Reply:
```suggestion
```